src/Npgsql/NpgsqlBinaryImporter.cs

```diff
@@ -358,6 +364,19 @@ async Task WriteRow(bool async, params object[] values)
                 await Write(value, async);
         }
 
+        /// <summary>
+        /// Checks if the column to be written belongs to the allowed columns for this importer
+        /// </summary>
+        /// <param name="targetColumnIndex">The target column index to be written. <see cref="_column"/></param>
+        /// <param name="currentNumColumns">The number of columns for this binary importer. <see cref="NumColumns"/></param>
+        void CheckColumnInTable(short targetColumnIndex, int currentNumColumns)
+        {
+            if (targetColumnIndex >= currentNumColumns)
+            {
+                throw new InvalidOperationException($"Could not write value. The target table does not contain a column at index {_column}");
```

Comment:
There is issue #2237 to move all exception throws to a helper. In this case it would allow the current method to be smaller than the limit on a method size which could be automatically inlined. If you're going to move this to the helper, it would be great!

Reply:
Sure, I can add to the `ThrowHelper` class, no problem! I just need some guidance:  Should it be a generic "OutOfBounds" exception, like: `ThrowInvalidOperationException_IndexOutOfBounds`, or something more specific for `NpgsqlBinaryImporter`?. As a first option (more generic) we could have something like:

```
[DoesNotReturn]
internal static void ThrowInvalidOperationException_IndexOutOfBounds(int index) =>
    throw new InvalidOperationException($"The index value {index} is outside the bounds of the array.");
```
What do you think?

Reply:
Let's write a specific message instead to help users to narrow issues in their code. Probably it should look like this, but feel free to improve it:

```csharp
[DoesNotReturn]
internal static void ThrowInvalidOperationException_BinaryImportParametersMismatch(int columnCount, int valueCount) =>
    throw new InvalidOperationException($"The binary import operation was started with {columnCount} column(s), but {valueCount} value(s) were provided.");
```

And this exception could be thrown from line 117 instead of this:

https://github.com/npgsql/npgsql/blob/4ee80f28a172f625e2f998186915c9efa7ea19a0/src/Npgsql/NpgsqlBinaryImporter.cs#L117